MultiLangProgGuide/MultiLangProgGuide.rst

```diff
@@ -1819,6 +1821,1220 @@ If the requested :code:`timeout_nsec` exceeds
+
+Go BufferT GetLenAlloc()
+------------------------
+
+.. code-block:: go
+
+    GetLenAlloc() uint
+
+Return the :code:`len_alloc` field of the :code:`C.ydb_buffer_t`
+structure referenced by :code:`cbuft` (zero if the structure has not
+yet been allocated, i.e., :code:`cbuft` is :code:`nil`).
+
+Go BufferT GetLenUsed()
+-----------------------
+
+.. code-block:: go
+
+    GetLenUsed() uint, int
+
+Return the :code:`len_used` field of the :code:`C.ydb_buffer_t`
+structure referenced by :code:`cbuft` as the first (:code:`uint`)
+return value (zero if the structure has not yet been allocated, i.e.,
+:code:`cbuft` is :code:`nil`).
+
+If the :code:`len_used` field of the :code:`C.ydb_buffer_t`
+structure is greater than the :code:`len_alloc` field (owing to a
+prior :code:`C.YDB_ERR_INVSTRLEN` error), the return code is
+:code:`C.YDB_ERR_INVSTRLEN`; otherwise, it is :code:`C.YDB_OK`.
+
+Go BufferT GetValBAry()
+-----------------------
+
+.. code-block:: go
+
+    GetValBAry() *[]byte, int
+
+If the :code:`len_used` field of the :code:`C.ydb_buffer_t` structure
+referenced by :code:`cbuft` is greater than its :code:`len_alloc`
+field (owing to a prior :code:`C.YDB_ERR_INVSTRLEN` error), return
+:code:`len_alloc` bytes of the buffer referenced by the
+:code:`C.ydb_buffer_t` structure referenced by :code:`cbuft` as a byte
+array, and a return code of :code:`C.YDB_ERR_INVSTRLEN`.
+
+Otherwise, return :code:`len_used` bytes of the buffer as a byte array
+(a zero length array if the structure has not yet been allocated,
+i.e., :code:`cbuft` is :code:`nil`), and a return code of
+:code:`C.YDB_OK`.
+
+Go BufferT GetValStr()
+----------------------
+
+.. code-block:: go
+
+    GetValStr() *string, int
+
+If the :code:`len_used` field of the :code:`C.ydb_buffer_t` structure
+referenced by :code:`cbuft` is greater than its :code:`len_alloc`
+field (owing to a prior :code:`C.YDB_ERR_INVSTRLEN` error), return
+:code:`len_alloc` bytes of the buffer referenced by the
+:code:`C.ydb_buffer_t` structure referenced by :code:`cbuft` as a
+string, and a return code of :code:`C.YDB_ERR_INVSTRLEN`.
+
+Otherwise, return :code:`len_used` bytes of the buffer as a string (a
+zero length string if the structure has not yet been allocated, i.e.,
+:code:`cbuft` is :code:`nil`), and a return code of :code:`C.YDB_OK`.
+
+Go BufferT SetValBAry()
+-----------------------
+
+.. code-block:: go
+
+    SetValBAry(val *[]byte) int
+
+
+If the length of :code:`val` is greater than the :code:`len_alloc`
+field of the :code:`C.ydb_buffer_t` structure referenced by
+:code:`cbuft`, make no changes and return :code:`C.YDB_ERR_INVSTRLEN`.
+
+Otherwise, copy the bytes of :code:`val` to the location referenced by
+the :code:`buf_addr` field of the :code:`C.ydbbuffer_t` structure, set
+the :code:`len_used` field to its length and return with a return code
+of :code:`C.YDB_OK`.
+
+Go BufferT SetValStr()
+----------------------
+
+.. code-block:: go
+
+    SetVarStr(val *string) int
```

Comment:
It seems to me that for BufferT and BufferTArray, `val` is preferable to `var` because we are setting and getting values from a buffer. `var` would be preferable for KeyT methods, except that we leave it blank there.